pkgs/by-name/de/dediprog-sf100/package.nix

```
@@ -0,0 +1,40 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, libusb
+, pkg-config
+}:
+
+stdenv.mkDerivation rec {
+  pname = "dediprog-sf100-linux";
+  version = "1.14.20.x";
+
+  src = fetchFromGitHub {
+    owner = "DediProgSW";
+    repo = "SF100Linux";
+    rev = "V${version}";
+    hash = "sha256-hQvBZIwaWEC41vj2flaekIUP9Fwtj/JPi3XwRxfUbD0=";
+  };
+
+  buildInputs = [ libusb ];
+  nativeBuildInputs = [ pkg-config ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    mkdir -p $out/share/DediProg
+    install -Dm0755 ./dpcmd -t $out/bin
+    install -Dm0644 ./ChipInfoDb.dedicfg -t $out/share/DediProg/ChipInfoDb.dedicfg
```

If you create some udev rules we could add it as well. For example:
```nix
udevRules = pkgs.writeText "dediprog.rules" 
    ''
       ACTION=="add", SUBSYSTEM=="usb", ATTRS{idVendor}=="0483", ATTRS{idProduct}=="dada", MODE="660", GROUP="plugdev"
    '';
```

Afterwards adding it to the nativeBuildInputs and copy it over in install phase:

```suggestion
    install -Dm0644 ./ChipInfoDb.dedicfg -t $out/share/DediProg/ChipInfoDb.dedicfg
    install -Dm644 ${udevRules}/dediprog.rules $out/lib/udev/rules.d/60-dediprog.rules
```

Maybe this is nice if someone want to use `services.udev.packages`.

Internally we have a nixos module for this udev rule.  

@sylv-io do you have some opinion?